Source/MSBuild.Sdk.Extras/Build/MSBuild.NET.Workarounds.targets

```diff
@@ -16,10 +16,10 @@
   -->
   <Target Name="_CheckAndFixupRootNamespaceWithGeneratedSuffix"
       BeforeTargets="_CompileTemporaryAssembly;CoreCompile"
-      Condition="('$(IsGeneratedProject)' == 'true') AND $([System.Text.RegularExpressions.Regex]::IsMatch('$(RootNamespace)', '$(GeneratedProjectSuffixPattern)'))">
+      Condition="'$(IsGeneratedProject)' == 'true' AND $([System.Text.RegularExpressions.Regex]::IsMatch('$(RootNamespace)', '$(_GeneratedProjectSuffixPattern)'))">
```

Comment:
How do I do it? Is it possible to store RegEx object in a MSBuild property?

Reply:
I'd store the results of a regex by storing the result in a property instead of using directly in a condition.

Reply:
Outside of the target right?

Reply:
Right, I think that would work. but you may not need it.

Reply:
If we don't set the `RootNamespace` it defaults to `MSBuildProjectName`. If it contains invalid chars or suffix for generated projects and if we dynamically transform existing `RootNamespace` (i.e. `MSBuildProjectName`) to the correct one, the suffix flows through. **From there all hell breaks loose! Believe me I've had my fair share**.

Reply:
I feel like this is better off as a StartsWith string comparison rather than a RegEx? Namespaces/Project Filenames contain "." characters which match any character in Regex plus this isn't using the "^" Start of String anchor, each potentially matching false positives. For example "My.Project.Name" would match "This.IsNot.My_Project_Namespace" in this scenario, which wouldn't be desired whereas "^My\.Project\.Name" would match things like "My.Project.Name.Sub.Namespace" (though it would also match "My.Project.Namespace").